types/go.mod

```diff
@@ -0,0 +1,18 @@
+module github.com/regen-network/regen-ledger/types
```

Comment:
Should we rename types to core?

If yes, maybe in a follow-up PR.

Reply:
cc @aaronc

Reply:
Follow up PR is fine

Reply:
@cyberbono3 could you add a new issue for that?